virtcontainers/vm.go

```diff
@@ -128,7 +128,7 @@ func NewVM(ctx context.Context, config VMConfig) (*VM, error) {
 	// 2. setup agent
 	agent := newAgent(config.AgentType)
 	vmSharePath := buildVMSharePath(id)
-	err = agent.configure(hypervisor, id, vmSharePath, isProxyBuiltIn(config.ProxyType), config.AgentConfig)
```

Comment:
This breaks vm factory because vm template needs to use a builtin proxy and the builtin flag and proxy url is not set. The factory code cannot call agent.init() because that call is bound with a sandbox. That's the reason I reinitialize agent's proxy fields in the `configure` method.

Reply:
It seems the vm factory only needs to set the builtin flag in the configure() method and
the proxy url will be set in the following setProxy(), so what I should do is just remove the 
proxy url initialization to fix this issue?

Reply:
Yes, I think so. Please make sure the PR passes kata-containers/tests#578 though.